src/main/java/io/javalin/http/JavalinResponseWrapper.kt

```diff
@@ -55,8 +76,47 @@ class OutputStreamWrapper(val res: HttpServletResponse, val rwc: ResponseWrapper
                 return // don't write body
             }
         }
-        write(resultStream.readBytes())
+        resultStream.copyTo(this)
         resultStream.close()
+        finalize()
+    }
+
+    private fun streamToOutput(b: ByteArray, off: Int, len: Int) {
+        when {
+            // Gzip only for now while streaming, as Brotli stream compression currently doesn't work
+            rwc.accepts.contains("gzip", ignoreCase = true) && rwc.compressionStrategy.gzip != null -> {
+                if (!res.containsHeader(Header.CONTENT_ENCODING)) {
+                    res.setHeader(Header.CONTENT_ENCODING, "gzip")
+                    if(gzipCompressor == null) {
```

Comment:
This check shouldn't be required? Also, can you make the compressor lateinit instead?